crates/ruff_python_ast/src/visitor/preorder.rs

```diff
@@ -476,8 +476,9 @@ where
             orelse,
             range: _range,
         }) => {
-            visitor.visit_expr(test);
+            // `body if test else orelse`
             visitor.visit_expr(body);
+            visitor.visit_expr(test);
```

Comment:
Some Python noob must have implemented the `PreOrder` visitor, assuming its the same as in JavaScript ;)